WalletWasabi.Fluent/ViewModels/Wallets/Home/Tiles/PrivacyRing/PrivacyBarViewModel.cs

```diff
@@ -89,69 +86,91 @@ private void CreateSegments(IExtendedList<PrivacyBarItemViewModel> list, IEnumer
 
 	private IEnumerable<PrivacyBarItemViewModel> CreateCoinSegments(IEnumerable<Pocket> pockets, int coinCount)
 	{
-		var total = pockets.Sum(x => Math.Abs(x.Amount.ToDecimal(NBitcoin.MoneyUnit.BTC)));
-		var start = 0.0m;
-
-		var usableWidth = Width - (coinCount * 2);
-
-		var usablePockets =
-				pockets.Where(x => x.Coins.Any())
-					   .OrderByDescending(x => x.Coins.First().HdPubKey.AnonymitySet)
-					   .ToList();
-
-		foreach (var pocket in usablePockets)
+		var totalAmount = pockets.Sum(x => Math.Abs(x.Amount.ToDecimal(MoneyUnit.BTC)));
+		var usableWidth = (decimal)Width - (coinCount - 1) * GapBetweenSegments;
+
+		// Calculate the width of the segments.
+		var rawSegments = pockets
+			.Where(x => x.Coins.Any())
+			.SelectMany(pocket => pocket.Coins)
+			.Select(coin =>
+			{
+				var pocket = pockets.First(pocket => pocket.Coins.Contains(coin));
+				var amount = coin.Amount.ToDecimal(MoneyUnit.BTC);
+				var width = Math.Abs(usableWidth * amount / totalAmount);
+
+				return (OwnerPocket: pocket, Coin: coin, Width: width);
```

Comment:
Line 98 is unnecesarily wasting CPU time searching for the pocket when you already had that.

Why not
```suggestion
			.SelectMany(pocket => pocket.Coins.Select(coin => 
			{
				var amount = coin.Amount.ToDecimal(MoneyUnit.BTC);
				var width = Math.Abs(usableWidth * amount / totalAmount);

				return (OwnerPocket: pocket, Coin: coin, Width: width);
			})).ToArray();
```

BTW this is the case where LINQ query syntax really shines:

```suggestion
		var rawSegments =
			from pocket in pockets
			from coin in pocket.Coins
			let amount = coin.Amount.ToDecimal(MoneyUnit.BTC)
			let width = Math.Abs(usableWidth * amount / totalAmount)
			select (OwnerPocket: pocket, Coin: coin, Width: width);
```

Is much more readable IMO.

Reply:
Committed the first one as the syntax of the second one is not common in the code base and I prefer consistency.

Nice btw, thanks.